opta/commands/apply.py

```diff
@@ -135,14 +142,27 @@ def _apply(
     Terraform.create_state_storage(layer)
     gen_opta_resource_tags(layer)
     if layer.cloud == "aws":
-        AWS(layer).upload_opta_config(config)
+        cloud_client = AWS(layer)
     elif layer.cloud == "google":
-        GCP(layer).upload_opta_config(config)
+        cloud_client = GCP(layer)
     elif layer.cloud == "azurerm":
-        Azure(layer).upload_opta_config(config)
+        cloud_client = Azure(layer)
     else:
         raise Exception(f"Cannot handle upload config for cloud {layer.cloud}")
 
+    previous_config: StructuredConfig = cloud_client.get_remote_config()
+    if previous_config is not None and "opta_version" in previous_config:
+        old_opta_version = previous_config["opta_version"]
+        if (
+            old_opta_version != DEV_VERSION
+            and semver.VersionInfo.parse(VERSION.strip("v")).compare(
+                old_opta_version.strip("v")
+            )
+            < 0
+        ):
+            raise UserErrors()
```

Comment:
Put some helpful message here?

Reply:
whoops, can't belice I forgot that one! All good now